code/controllers/subsystem/processing/greyscale.dm

```diff
@@ -107,4 +116,66 @@ PROCESSING_SUBSYSTEM_DEF(greyscale)
 	for(var/color in 2 to length(split_colors))
 		. += "#[split_colors[color]]"
 
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviews()
+	// Put subtypes before their parent or the parent file will take all the generated icons
+	var/static/list/types_that_get_their_own_file = list(
+		"turfs" = /turf, // None of these yet but it's harmless to be prepared
+		"mobs" = /mob, // Ditto
+		"clothing/accessory" = /obj/item/clothing/accessory,
+		"clothing/head" = /obj/item/clothing/head,
+		"clothing/mask" = /obj/item/clothing/mask,
+		"clothing/neck" = /obj/item/clothing/neck,
+		"clothing/shoes" = /obj/item/clothing/shoes,
+		"clothing/suit" = /obj/item/clothing/suit,
+		"clothing/under" = /obj/item/clothing/under,
+		"clothing/clothing" = /obj/item/clothing,
+		"items/encryptionkey" = /obj/item/encryptionkey,
+		"items/pda" = /obj/item/modular_computer/pda,
+		"items/item" = /obj/item,
+		"objects" = /obj,
+)
+
+	var/list/handled_types = list()
+	for(var/filename in types_that_get_their_own_file)
+		var/type_to_export = types_that_get_their_own_file[filename]
+		handled_types += ExportMapPreviewsForType(filename, type_to_export, handled_types)
+
+	ExportMapPreviewsForType("unsorted", /atom, handled_types)
+
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviewsForType(filename, atom/atom_typepath, list/type_blacklist)
+	var/list/handled_types = list()
+	var/list/icons = list()
+	for(var/atom/fake as anything in subtypesof(atom_typepath))
+		if(type_blacklist && type_blacklist[fake])
+			continue
+		handled_types[fake] = TRUE
+		var/greyscale_config = fake::greyscale_config
+		var/greyscale_colors = fake::greyscale_colors
+		if(!greyscale_config || !greyscale_colors || fake::does_not_generate_gags_preview)
+			continue
+		var/icon/map_icon = icon(GetColoredIconByType(greyscale_config, greyscale_colors))
+		if((map_icon.Height() > 32) || (map_icon.Width() > 32)) // No large icons, use icon_preview and icon_preview_state instead.
```

Comment:
It's probably best to limit this type of checking to unit tests and make devs mark any non 32x32 icons with `does_not_generate_gags_preview`. Unnecessarily loading a large volume of icons is pretty slow.

Same deal for using `IconStates()`. If you port this to iconforge spritesheet generation, it will spit out the correct, descriptive, errors during CI and devs will fix it then. No need to check it beforehand.

Reply:
You would only skip this check if it was IconForge powered. As it is, this should be limited to unit tests since devs depend on that to catch the diffs anyway.